src/dodal/devices/util/lookup_tables_apple2.py

```diff
@@ -357,10 +370,30 @@ def update_lookuptables(self):
         """
         Update lookup tables from files and validate their format.
         """
-        LOGGER.info("Updating lookup table from file for gap.")
+        LOGGER.info("Updating lookup table for gap.")
         self._update_gap_lut()
-        LOGGER.info("Updating lookup table from file for phase.")
-        self._update_phase_lut()
+        if self.lut_config.phase_path is None:
+            LOGGER.info("Generating lookup table for phase.")
+            self._generate_phase_lut()
+
+        else:
+            LOGGER.info("Updating lookup table for phase.")
+            self._update_phase_lut()
+
+    def _generate_phase_lut(self):
+        for key in self.lookup_tables.gap.root.keys():
+            if key is not None:
+                self.lookup_tables.phase.root[Pol(key.lower())] = (
+                    generate_lookup_table_entry(
+                        min_energy=self.lookup_tables.gap.root[
+                            Pol(key.lower())
+                        ].limit.minimum,
+                        max_energy=self.lookup_tables.gap.root[
+                            Pol(key.lower())
+                        ].limit.maximum,
+                        poly1d_param=(PhasePoly1dParameters[Pol(key.lower())]),
+                    )
+                )
```

Comment:
```suggestion
        for key in self.lookup_tables.gap.root.keys():
            if key is not None:
                self.lookup_tables.phase.root[key] = (
                    generate_lookup_table_entry(
                        min_energy=self.lookup_tables.gap.root[
                            key
                        ].limit.minimum,
                        max_energy=self.lookup_tables.gap.root[
                            key
                        ].limit.maximum,
                        poly1d_param=(PhasePoly1dParameters[key]),
                    )
                )
```


I personally think this is overcomplicating the standard `EnergyMotorLookup` class. If we need specialised behaviour for the `EnergyMotorLookup`, we should be creating another class that extends / overrides the logic. 

So for i09's case, we should create J09EnergyMotorLookup which overrides `_update_phase_lut` to provide the custom phase values. 

I don't even think we need to do this either though because we should be able to just add these limits inside a lookup table file like the rest? (correct me if I'm wrong).

Reply:
I think it is ok to establish a default phase lookup table within the base class, particularly since this default value is functional for all Apple II  and most beamlines currently lack their own phase lookup table.